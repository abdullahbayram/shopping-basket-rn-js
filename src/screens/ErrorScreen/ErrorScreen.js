import React from 'react';
import { useRoute } from '@react-navigation/native';
import RedirectWithAnimation from '../../components/templetes/RedirectWithAnimation';
import Screen from '../../components/templetes/BaseScreen';
import strings from '../../constants/strings';

const ErrorScreen = () => {
  const route = useRoute();
  const { errorMessage } = route.params || { errorMessage: strings.unexpectedErrorShort };

  return (
    <Screen>
      <RedirectWithAnimation message={errorMessage} duration={10000} />
    </Screen>
  );
};

export default ErrorScreen;
