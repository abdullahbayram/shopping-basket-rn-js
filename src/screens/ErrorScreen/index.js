import React from 'react';
import { useRoute } from '@react-navigation/native';
import { BaseScreen, RedirectWithAnimation } from '@components/templates';
import { strings } from '@constants';

const ErrorScreen = () => {
  const route = useRoute();
  const { errorMessage } = route.params || { errorMessage: strings.payment.unexpectedErrorShort };

  return (
    <BaseScreen>
      <RedirectWithAnimation message={errorMessage} duration={10000} />
    </BaseScreen>
  );
};

export default ErrorScreen;
