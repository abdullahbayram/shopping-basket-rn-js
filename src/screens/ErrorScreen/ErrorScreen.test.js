import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { useRoute } from '@react-navigation/native';
import ErrorScreen from './ErrorScreen';

// Mock useRoute to control the route parameters for testing
jest.mock('@react-navigation/native', () => ({
  useRoute: jest.fn(),
}));

describe('ErrorScreen Component', () => {
  it('should render the default error message when no parameters are passed', () => {
    useRoute.mockReturnValue({
      params: {},
    });

    render(<ErrorScreen />);

    expect(screen.getByText('Unexpected Error')).toBeTruthy();
  });

  it('should render a custom error message when provided via navigation params', () => {
    useRoute.mockReturnValue({
      params: { errorMessage: 'Custom Error Message' },
    });

    render(<ErrorScreen />);

    expect(screen.getByText('Custom Error Message')).toBeTruthy();
  });

  it('should match the snapshot', () => {
    useRoute.mockReturnValue({
      params: { errorMessage: 'Snapshot Error Test' },
    });

    const { toJSON } = render(<ErrorScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
