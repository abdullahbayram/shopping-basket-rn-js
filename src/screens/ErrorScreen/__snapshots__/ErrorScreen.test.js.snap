// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen Component should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            undefined,
          ],
        ]
      }
    >
      Unexpected Error
    </Text>
  </View>
</View>
`;
